packages/ui-components/src/Input/CommonInput.tsx

```diff
@@ -1,26 +1,28 @@
 import React from "react";
 import classNames from "classnames/bind";
 import "./input.module.scss";
-import { FieldRenderProps, FieldMetaState } from "react-final-form";
 
-export interface CommonInputProps {
+export interface CommonInputProps
+  extends React.DetailedHTMLProps<
+    React.InputHTMLAttributes<HTMLInputElement>,
+    HTMLInputElement
+  > {
   // this should be the element containing the input
   fieldInput?: JSX.Element;
-  className?: string;
   help?: string | JSX.Element;
   // derived from final form type
-  // error is assigned a value of meta.error
-  // which has type any
+  // error is assigned a value of meta.error which has type any
   // eslint-disable-next-line @typescript-eslint/no-explicit-any
   error?: any;
   inline?: boolean;
-  // FieldValue passed to meta is string for all TextTypeInputs
-  // boolean for CheckboxInput, and number for NumberInput
-  meta?: FieldMetaState<string | number | boolean>;
```

Comment:
Handled by `TextAndNumberProps`, since that extends FieldRenderProps, which defines meta and input